api/app/signals/apps/signals/models/category_assignment.py

```diff
@@ -17,6 +18,16 @@ class CategoryAssignment(CreatedUpdatedModel):
 
     extra_properties = JSONField(null=True)  # TODO: candidate for removal
```

Comment:
Perhaps the `extra_properties` can be used to store the deadlines as a JSON blob, like you suggested earlier this week. Just to point out that this could be the place, not something we need to address in this PR (but I just wanted to mention it)

Reply:
I'll document the new feature with the caveat that it may still be tweaked.